src/wp-includes/class-wp-block-supports.php

```diff
@@ -155,6 +167,37 @@ private function register_attributes() {
 			}
 		}
 	}
+
+	/**
+	 * Modify the rendered block content with additional filters.
+	 *
+	 * @param  string $block_content Rendered block content.
+	 * @param  array  $block         Block object.
+	 * @return string                Filtered block content.
+	 */
+	public function render_block_supports( $block_content, $block ) {
+		$block_type       = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+		$block_attributes = $block['attrs'];
+
+		if ( ! $block_type ) {
+			return $block_content;
+		}
+
+		foreach ( $this->block_supports as $name => $block_support_config ) {
+			if ( ! isset( $block_support_config['render_block'] ) ) {
+				continue;
+			}
+
+			$block_content = call_user_func(
+				$block_support_config['render_block'],
```

Comment:
So far the block supports were only used to actually add classnames or inline styles to the wrapper, this PR is changing that by allowing them to register a generic hook for render_block. 

The use-case for these is to potentially support static blocks (adding style and classes to  static blocks that don't explicitly call the `get_block_wrapper_attributes` but also to allow adding style tags in addition to the block markup. These are needed by two Gutenberg PRs. https://github.com/WordPress/gutenberg/pull/26752 and https://github.com/WordPress/gutenberg/pull/29335

It's not entirely clear why this hook is better than just using `render_block` directly though.

Curious though about your thoughts here @mcsf @mtias

Reply:
A bit late... but I see that the parent PR (https://github.com/WordPress/gutenberg/pull/26752) is progressing nicely without this. Yes, I'm more in favour of not adding a new indirect API, just recommend `render_block`. If later we find that we have more compelling use cases for custom server rendering in block-supports, we can reassess.